lib/ClangImporter/ImporterImpl.h

```diff
@@ -552,6 +552,10 @@ class LLVM_LIBRARY_VISIBILITY ClangImporter::Implementation
   /// Mapping of already-imported declarations.
   llvm::DenseMap<std::pair<const clang::Decl *, Version>, Decl *> ImportedDecls;
 
+  // Types that should be imported as non-escapable. This is collected from
+  // the using declarations and used when importing template specializations.
+  llvm::DenseSet<const clang::Type *> nonEscapableTypes;
```

Comment:
I think this `DenseSet` only stores a small part of the non-escapable types (the ones that are typedef-ed), is that right? If so, I think it would be better to reflect that in the name of the field.

Reply:
Good point, will do!